### PR TITLE
new PB115 validate

### PR DESCRIPTION
**Pull request by aaron1535**
## Related Issues
fixes:

## Description
Adaptation of validation PB115 to the new validator

### CODE REVIEW

File: demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md [hunk 1]
```diff
@@ -43,6 +43,27 @@ Contributions are welcome and appreciated. For more info, visit our [Contributio
 Contributions are welcome and appreciated. For more info, visit our [Contribution Guide](https://xsoar.pan.dev/docs/contributing/contributing).
 ### Pack Contributors:
 
+---
```
Comment (YuvHayun) [inline demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md:46]: revert changes here.

File: Tests/id_set.json [hunk 2]
```diff
@@ -0,0 +1,90 @@
+{
+    "scripts": [],
```
Comment (YuvHayun) [inline Tests/id_set.json:2]: remove this file.

File: demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    ValidationResult,
+)
+
+ContentTypes = Playbook
+
+
+class IsTasksQuietModeValidator(BaseValidator[ContentTypes]):
+    error_code = "PB115"
+    description = "Checks if all tasks in a playbook are in quiet mode."
+    error_message = "Playbook '{playbook_name}' contains tasks that are not in quiet mode (quietmode: 2)."
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py:19]: what about rationale?

File: demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+            ValidationResult(
+                validator=self,
+                message=self.error_message,
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if (
+                any(
+                    (i.get("playbookInputQuery") or {}).get("queryEntity") == "indicators"
+                    for i in content_item.data.get("inputs", {})
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py:45]: rename i to inputs, more readable and meaningful.

File: demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+        return [
+            ValidationResult(
+                validator=self,
+                message=self.error_message,
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if (
+                any(
+                    (i.get("playbookInputQuery") or {}).get("queryEntity") == "indicators"
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py:44]:
```suggestion
                    (i.get("playbookInputQuery", {}).get("queryEntity") == "indicators"
```

File: demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+                    for i in content_item.data.get("inputs", {})
+                )
+                and [
+                    task_key
+                    for _, task_key in content_item.tasks.items()
+                    if task_key.quietmode == 2
+                ]
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py:51]: Let's return the ids of the tasks that uses quite mode to make the message error easier for the user to fix.
Also, let's save the tasks so we can reuse it in the fix function.
See Sapir's [pr](https://github.com/demisto/demisto-sdk/pull/4360/files#diff-789dba8ca3e318b657b72a8662bf27bbb0a162501791238c7410db2bedeab72f) as a good reference.

**Review by YuvHayun — CHANGES_REQUESTED** [review]
See my comments.
Also, don't forget to add changelog and the error code to the config file.

File: Tests/id_set.json (hunk 2 above)
Comment (aaron1535) [reply Tests/id_set.json:2]: done

File: demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md (hunk 1 above)
Comment (aaron1535) [reply demisto_sdk/tests/test_files/content_slim/Packs/Sample01/README.md:46]: done

File: demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py (hunk 4 above)
Comment (aaron1535) [reply demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py:45]: changed to input

File: demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py (hunk 5 above)
Comment (aaron1535) [reply demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py:44]: done

File: demisto_sdk/commands/validate/sdk_validation_config.toml [hunk 7]
```diff
@@ -29,8 +29,12 @@ select = [
     "IN100", "IN101", "IN102", "IN104", "IN106", "IN107", "IN108", "IN109", "IN110", "IN112", "IN113", "IN114", "IN115", "IN117", "IN118", "IN121", "IN122", "IN123", "IN124", "IN125", "IN126", "IN127", "IN130",
     "IN131", "IN134", "IN135", "IN139", "IN141", "IN142", "IN144", "IN145", "IN146", "IN149", "IN150", "IN151", "IN152", "IN153", "IN154", "IN156", "IN158", "IN159", "IN160",
     "IN161", "IN162",
-    "PB100", "PB101", "PB103", "PB104", "PB105", "PB108", "PB118", "PB122", "PB123", "PB125", "PB126",
+
+    
+
+    "PB100", "PB101", "PB103", "PB104", "PB105", "PB108", ,"PB115", "PB118", "PB122", "PB123", "PB125", "PB126",
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/sdk_validation_config.toml:35]:
```suggestion
"PB100", "PB101", "PB103", "PB104", "PB105", "PB108", "PB115", "PB118", "PB122", "PB123", "PB125", "PB126",
```

File: demisto_sdk/commands/validate/sdk_validation_config.toml [hunk 8]
```diff
@@ -29,8 +29,12 @@ select = [
...
     "PB100", "PB101", "PB103", "PB104", "PB105", "PB108", "PB118", "PB122", "PB123", "PB126", "PB106",
+
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/sdk_validation_config.toml:37]: remove

File: demisto_sdk/commands/validate/sdk_validation_config.toml [hunk 9]
```diff
@@ -53,8 +57,10 @@ select = [
     "IN131", "IN134", "IN135", "IN139", "IN141", "IN142", "IN144", "IN145", "IN146", "IN149", "IN150", "IN151", "IN152", "IN153", "IN154", "IN156", "IN158", "IN159", "IN160",
     "IN161", "IN162",
     "LO107",
-    "PB100", "PB101", "PB103", "PB104", "PB105", "PB108", "PB122", "PB123", "PB125", "PB126",
+
+    "PB100", "PB101", "PB103", "PB104", "PB105", "PB108", "PB115", "PB122", "PB123", "PB125", "PB126",
     "PB100", "PB101", "PB103", "PB104", "PB105", "PB108", "PB122", "PB123", "PB126", "PB106",
+
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/sdk_validation_config.toml:63]: remove

File: demisto_sdk/commands/validate/sdk_validation_config.toml [hunk 10]
```diff
@@ -53,8 +57,10 @@ select = [
     "IN131", "IN134", "IN135", "IN139", "IN141", "IN142", "IN144", "IN145", "IN146", "IN149", "IN150", "IN151", "IN152", "IN153", "IN154", "IN156", "IN158", "IN159", "IN160",
     "IN161", "IN162",
     "LO107",
-    "PB100", "PB101", "PB103", "PB104", "PB105", "PB108", "PB122", "PB123", "PB125", "PB126",
+
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/sdk_validation_config.toml:60]: remove

File: demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py [hunk 11]
```diff
@@ -0,0 +1,98 @@
...
+    error_message = "Playbook '{playbook_name}' contains tasks that are not in quiet mode (quietmode: 2) The tasks names is: '{tasks}'."
+    fix_message = (
+        "Fixed playbook '{playbook_name}' to set tasks '{tasks}' to (quietmode: 0)."
+    )
+    related_field = "tasks"
+    is_auto_fixable = True
+    related_file_type = [RelatedFileType.YML]
+    invalid_tasks_in_playbooks = {}
+
+    def get_invalid_task_ids(self, content_item: ContentTypes) -> List:
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py:29]:
```suggestion
    def get_invalid_task_ids(self, content_item: Playbook) -> List:
```

File: demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py [hunk 12]
```diff
@@ -0,0 +1,98 @@
...
+        Returns:
+            List[str]: List of task IDs where quietmode == 2.
+        """
+        invalid_task_ids = [
+            task
+            for task in content_item.tasks.values()
+            if task.quietmode == 2
+        ]
+        if invalid_task_ids:
+            self.invalid_tasks_in_playbooks[content_item.name] = invalid_task_ids
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py:45]:
```suggestion
        self.invalid_tasks_in_playbooks[content_item.name] = invalid_task_ids
```
We don't mind writing it in any case, worst case we'll write an empty list into it.

File: demisto_sdk/commands/validate/tests/PB_validators_test.py [hunk 13]
```diff
@@ -327,6 +331,117 @@ def test_IsAskConditionHasUnhandledReplyOptionsValidator():
...
+                "id": "test task",
+                "taskid": "27b9c747-b883-4878-8b60-7f352098a631",
+                "type": "condition",
+                "message": {"replyOptions": ["yes"]},
+                "nexttasks": {"no": ["1"]},
+                "task": {"id": "27b9c747-b883-4878-8b60-7f352098a63c"},
+                "quietmode": 1,
+            }
+        )
+    }
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/PB_validators_test.py:439]: you can put it inside create_playbook_object arguments.

File: demisto_sdk/commands/validate/tests/PB_validators_test.py [hunk 14]
```diff
@@ -327,6 +331,117 @@ def test_IsAskConditionHasUnhandledReplyOptionsValidator():
...
+    validator = IsTasksQuietModeValidator()
+    assert len(validator.is_valid([playbook])) == 0
+    fix_playbook = validator.fix(playbook).content_object
+    assert fix_playbook == playbook
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/PB_validators_test.py:443]: Not sure what you tried to do here?
If line 441 return 0 then what are we fixing?

File: demisto_sdk/commands/validate/tests/PB_validators_test.py [hunk 15]
```diff
@@ -327,6 +331,117 @@ def test_IsAskConditionHasUnhandledReplyOptionsValidator():
...
+                "id": "test fail task No2",
+                "taskid": "27b9c747-b883-4878-8b60-7f352098a631",
+                "type": "condition",
+                "message": {"replyOptions": ["yes"]},
+                "nexttasks": {"no": ["1"]},
+                "task": {"id": "27b9c747-b883-4878-8b60-7f352098a63c"},
+                "quietmode": 2,
+            }
+        ),
+    }
```
Comment (YuvHayun) [inline demisto_sdk/commands/validate/tests/PB_validators_test.py:386]: Let's make it create_playbook_object args.

**Review by YuvHayun — APPROVED** [review]
Great job! see my comments.
Approved from my side.

File: demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py [hunk 16]
```diff
@@ -0,0 +1,94 @@
...
+                validator=self,
+                message=self.error_message.format(
+                    playbook_name=content_item.name,
+                    tasks=", ".join([task.id for task in invalid_tasks]),
+                ),
+                content_object=content_item,
+            )
+            for content_item in content_items
+            if (
+                any(
```
Comment (JudahSchwartz) [inline demisto_sdk/commands/validate/validators/PB_validators/PB115_is_tasks_quiet_mode.py:65]: @talzich wrote a beautful helper function in his pr. idealy lets use the same one

**Review by JudahSchwartz — APPROVED** [review]
fantastic, note my comment

**Pull request merged**